### PR TITLE
Refactor ntpd with chronyd

### PR DESCRIPTION
**Pull request by jcnars**
ntpd has been deprecated in favor of chronyd from EL 8.x
Chronyd has been around for some time as a functionally superior alternative for ntpd. 

With this change, we are uniformly using chrony to implement Oracle's NTP synchronization.
The proposed code changes have been tested successfully (internal: http://b/239771476) on an EL8.5 OEL host against 19c.

Testing it against a RHEL7.7 host with the prowjob with this PR.

### CODE REVIEW

**Comment by jcnars** [comment]
#### tl;dr:
* A good part of the day today was spent in ensuring that the code changes proposed:
  * not only work in a forward-compatible setup (EL 8.5 for Oracle 19c), 
  * but also in a backward-compatible setup (EL 7.7 for Oracle versions 12.2, 18c & 19c). 
* This level of testing was discussed with @mfielding  prior to the weekend.

##### tl;dr result:
* The code changes did work flawlessly for EL 7.7 for Oracle versions 12.2, 18c & 19c !

**Review by mfielding — APPROVED** [review]
Although I had issues getting a default chrony.conf to update, when I try again I can't reproduce the issue:

# mv /etc/chronyy.conf /etc/chrony.conf.backup
# yum -y reinstall chrony
[root@linuxserver44 mfielding]# grep server /etc/chrony.conf
# service chronyd restart
# date
Thu Aug 25 07:32:22 PDT 2022
# grep server /etc/chrony.conf
server 0.rhel.pool.ntp.org iburst
server 1.rhel.pool.ntp.org iburst
server 2.rhel.pool.ntp.org iburst

- run the prep steps only (with --prep-host and --ntp-pref)

$ sh ./install-oracle.sh --ora-swlib-bucket gs://bmaas-testing-oracle-software --instance-ssh-user ansible --instance-ssh-key ~/.ssh/id_rsa --backup-dest /u01/backups --ora-swlib-path /u01/oracle_install --ora-version 12.2 --ora-swlib-type gcs --ora-asm-disks syd1_asm.json --ora-data-mounts syd1_datamounts.json --cluster-type RAC --cluster-config syd1_cluster.json --ora-data-diskgroup DATA --ora-reco-diskgroup RECO --ora-db-name orcl --ora-db-container false --ntp-pref=10.210.1.7 --prep-host > test-prep.out

[root@linuxserver44 ~]# chronyc tracking
Reference ID    : 0AD20107 (gateway-instance)
Stratum         : 4
Ref time (UTC)  : Thu Aug 25 14:34:19 2022
System time     : 0.000000016 seconds slow of NTP time


Testing with ntpd installed seems to work too:

service chronyd stop
yum remove chrony
yum install ntp
echo "server 10.210.1.7 iburst prefer" >> /etc/ntp.conf
service ntpd start

# ntpstat
synchronised to NTP server (10.210.1.7) at stratum 4
   time correct to within 290 ms
   polling server every 64 s

(ru

[root@linuxserver44 ~]# ntpstat
bash: /bin/ntpstat: No such file or directory
[root@linuxserver44 ~]# chronyc tracking
Reference ID    : 0AD20107 (gateway-instance)
Stratum         : 4
Ref time (UTC)  : Thu Aug 25 14:42:57 2022
System time     : 0.000029745 seconds slow of NTP time
Last offset     : -0.000090340 seconds

$ sh ./install-oracle.sh --ora-swlib-bucket gs://bmaas-testing-oracle-software --instance-ssh-user ansible --instance-ssh-key ~/.ssh/id_rsa --backup-dest /u01/backups --ora-swlib-path /u01/oracle_install --ora-version 12.2 --ora-swlib-type gcs --ora-asm-disks syd1_asm.json --ora-data-mounts syd1_datamounts.json --cluster-type RAC --cluster-config syd1_cluster.json --ora-data-diskgroup DATA --ora-reco-diskgroup RECO --ora-db-name orcl --ora-db-container false --ntp-pref=10.210.1.7 --prep-host > test-prep.out &


[root@linuxserver44 ~]# ntpstat
bash: /bin/ntpstat: No such file or directory
[root@linuxserver44 ~]# chronyc tracking
Reference ID    : 0AD20107 (gateway-instance)
Stratum         : 4
Ref time (UTC)  : Thu Aug 25 14:42:57 2022
System time     : 0.000029745 seconds slow of NTP time
Last offset     : -0.000090340 seconds

I also managed to get a 12.2 RAC install to run without NTP complaints.
